nixos/modules/services/mail/dovecot.nix

```
@@ -1,19 +1,90 @@
 { config, lib, pkgs, ... }:
 
 let
-  inherit (lib) attrValues concatMapStringsSep concatStrings
-    concatStringsSep flatten imap1 literalExpression mapAttrsToList
-    mkEnableOption mkIf mkOption mkRemovedOptionModule optional optionalAttrs
-    optionalString singleton types mkRenamedOptionModule nameValuePair
-    mapAttrs' listToAttrs filter;
-  inherit (lib.strings) match;
+  inherit (lib) attrValues concatMapStringsSep concatStrings concatStringsSep
+    flatten imap1 isBool isDerivation isInt isList isPath isString
+    literalExpression mapAttrsToList mkEnableOption mkIf mkOption
+    mkRemovedOptionModule optional optionalAttrs optionalString singleton
+    types mkRenamedOptionModule nameValuePair mapAttrs' listToAttrs filter;
+  inherit (lib.generators) toPretty;
+  inherit (lib.lists) findFirstIndex length sublist reverseList;
+  inherit (lib.strings) escape match splitString;
 
   cfg = config.services.dovecot2;
   dovecotPkg = pkgs.dovecot;
 
   baseDir = "/run/dovecot2";
   stateDir = "/var/lib/dovecot";
 
+  /* Helper: Format string in way that may strech over multiple
+   * lines in a Dovecot configuration file
+   *
+   * This is a little complicated since Dovecot does not like when the last
+   * line is empty, so we first have to truncate all trailing empty lines
+   * before when can join them using the continuation line syntax.
+   */
+  formatDovecotMultiline =
+    string:
+      let
+        # Split input into line list
+        lines = splitString "\n" string;
+        # Find index of last non-empty line in line list
+        lastContentIndex = length lines - (findFirstIndex (
+          line: match "[[:space:]]*" line == null  # <=> Line is not empty
+        ) (length lines) (reverseList lines));
+        # Drop empty trailing lines from line list
+        trimmedLines = sublist 0 lastContentIndex lines;
+      in
+        # Reassemble content without trailing lines but with continuation markers
+        concatStringsSep " \\\n" trimmedLines;
+
+  /* Helper: Apply string escaping and quotation to Dovecot string if needed
+   *
+   * Dovecot is sometimes touchy about strings being quoted, so this only
+   * quotes the string when it is actually needed.
+   */
+  escapeDovecotString = string:
+    let
+      escapedString = escape ["\\" "\""] string;
+    in   if match "[[:space:]]*<.*|.*[[:space:]#\"\\].*|" string != null
```

```suggestion
    in   if match ''[[:space:]]*<.*|.*[[:space:]#"\].*|'' string != null
```

so you can copy it verbatim